lib/client.c

```diff
@@ -131,29 +133,38 @@ char *get_query_metadata(SF_CONNECT *sf, const char *query_id) {
     return metadata;
   }
   SF_FREE(status_query);
-  log_trace("Error getting query metadata.");
+  log_error("Error getting query metadata. Query id: %s", sfstmt->sfqid);
   return NULL;
 }
 
-/**
- * Get the status of the query
- * @param sf the SF_CONNECT context
- * @param query_id the query id
- */
-SF_QUERY_STATUS get_query_status(SF_CONNECT *sf, const char *query_id) {
+
+SF_QUERY_STATUS snowflake_get_query_status(SF_STMT *sfstmt) {
   SF_QUERY_STATUS ret = SF_QUERY_STATUS_NO_DATA;
-  char *metadata = get_query_metadata(sf, query_id);
+  char *metadata = get_query_metadata(sfstmt);
   if (metadata) {
     cJSON* metadataJson = snowflake_cJSON_Parse(metadata);
 
     cJSON* status = snowflake_cJSON_GetObjectItem(metadataJson, "status");
-    if (snowflake_cJSON_IsString(status))
-    {
+    if (snowflake_cJSON_IsString(status)) {
       char* queryStatus = snowflake_cJSON_GetStringValue(status);
       ret = get_status_from_string(queryStatus);
     }
+    else {
```

Comment:
After seeing the logic in `get_real_results` I'm not quite sure whether we should set error here. Seems no data is acceptable, the error log in `get_query_metadata` would be the same.